src/inverted-index.js

```diff
@@ -0,0 +1,160 @@
+      } else {
+        Object.keys((index[fileName].index)).forEach((token) => {
+          // check content of token indices
+          if (!((index[fileName].index[token]) instanceof Array)) {
+            validIndex = false;
+          } else {
+            (index[fileName].index[token]).forEach((digit) => {
+              if ((typeof digit) !== 'number') {
+                validIndex = false;
+                return false;
+              }
+            });
+          }
+        });
+      }
+    });
+    if (!validIndex) {
+      return { error: 'invalid index' };
+    } else
+    // Check to see if file name was specified
+    if (this.hasFileName(terms)) {
+      const fileIndex = {};
+      const fileName = terms[0];
+      // Check if index exists for file name
+      if (index[fileName] === undefined) {
+        return { error: 'no index created for that book' };
+      }
+      // Check if index has valid content for that file name
+      const indexForFile = index[fileName].index;
+      if (indexForFile === undefined) {
+        return { error: 'no index created for that book' };
+      }
+      const searchTerms = terms.slice(1, terms.length);
+      this.getTokens(searchTerms, tokens);
+      tokens.forEach((token) => {
+        if (indexForFile[token] === undefined) {
+          fileIndex[token] = [];
+        } else {
+          fileIndex[token] = indexForFile[token];
+        }
+      });
+      result[fileName] = fileIndex;
+    } else {
+      // Loop through index object for all files
+      Object.keys(index).forEach((fileName) => {
+        const fileIndex = {};
+        // Skip files with error
+        if (Object.keys(index[fileName])[0] === 'error') {
+          return; // Continue to next file
+        }
+        const indexForFile = index[fileName].index;
+        const searchTerms = terms;
+        this.getTokens(searchTerms, tokens);
+        tokens.forEach((token) => {
+          if (indexForFile[token] === undefined) {  // When the term is not in the book
+            fileIndex[token] = [];
+          } else {
+            fileIndex[token] = indexForFile[token];
+          }
+        });
+        result[fileName] = fileIndex;
+      });
+    }
+    return result;
+  }
+    /**
+   * This method flattens embedded arrays into a single array
+   * @param {Array} items An array to be flattened
+   * @param {*} flattened The flattened array, filled by closure
+   * @returns {null} returns null
+   */
+  getTokens(items, filteredTokens) {
+    items.forEach((element) => {
+      if (Array.isArray(element)) {
+        this.getTokens(element, filteredTokens);
+      } else if (element.indexOf(' ') >= 0) { // Check for words with spaces
+        const spacesRemoved = element.split(/\s+/);
+        this.getTokens(spacesRemoved, filteredTokens);
+      } else {
+        filteredTokens.push(element);
+      }
+    });
+  }
+
+  /**
+   * This method checks if the first term in the array is a file name
```

Comment:
Oh thanks. I've effected that. Thanks a lot for your contributions to my project.